crate/operator/handlers/handle_ping_cratedb_status.py

```diff
@@ -57,23 +59,39 @@ async def ping_cratedb_status(
         patch.status[CLUSTER_STATUS_KEY] = {"health": "SUSPENDED"}
         return
 
-    async with GlobalApiClient() as api_client:
-        core = CoreV1Api(api_client)
-        host = await get_host(core, namespace, name)
-        password = await get_system_user_password(core, namespace, name)
-    conn_factory = connection_factory(host, password)
-
     try:
-        async with conn_factory() as conn:
-            async with conn.cursor() as cursor:
+        async with GlobalApiClient() as api_client:
+            core = CoreV1Api(api_client)
+            host = await get_host(core, namespace, name)
+            password = await get_system_user_password(core, namespace, name)
+        conn_factory = connection_factory(host, password)
+        connection = conn_factory()
+
+        async with connection as conn:
+            cursor_cm = await conn.cursor()
+            async with cursor_cm as cursor:
                 healthiness = await get_healthiness(cursor)
                 # If there are no tables in the cluster, get_healthiness returns
                 # none: default to `Green`, as cluster is reachable
                 status = HEALTHINESS_TO_STATUS.get(
                     healthiness, PrometheusClusterStatus.GREEN
                 )
     except Exception as e:
-        logger.warning("Failed to ping cluster.", exc_info=e)
+        match e:
```

Comment:
do you know what is the most efficient between using `match` and many `except` block?

Reply:
or maybe you can build the custom message in the cases and after the `match`, have `logger.warning(f"{error_msg}", e)` just once.

Reply:
hm, you're right. match-case is probably not the most efficient solution here. I changed it to a plain except with isinstance checks to keep it simple. Still keeping the unexpected errors separate because in the case of an unexpected error, we still want to log the whole stacktrace